uwblueprint/focus-on-nature
# feat: create registration API

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-38b49f83f6ce4b07ae188dabb37631e8?p=7959b895c46d4060a99dfd47771096a2)



## Implementation description
* Added `/camper/register` route which allows camper registration, which adds a new camper to the database and also updates the `campers` field in the camp that was registered for
* Added camper validator



## Steps to test
1. Test manually in Postman - create a POST request to /camper/register with Body containing required valid JSON data and ensure no errors in console logs, and also that the new camper is added to the database + camp is updated
2. Test a camper with a `camp` field that does not exist - should throw an error



## What should reviewers focus on?
* Verify the validator and main implementation in `backend/typescript/services/implementations/camperService.ts`


## Checklist
